src/index.js

```diff
@@ -1,23 +1,84 @@
+/* @flow */
 /* eslint prefer-template: 0 */
 import React from 'react';
 import accepts from 'attr-accept';
 import getDataTransferItems from './getDataTransferItems';
 
+type InputAttributes = {
+  accept: string,
+  type: string,
+  style: Object,
+  multiple: boolean,
+  name?: string,
+  ref: Function,
+  onChange: Function
+};
+
+export type Props = {
+  onClick: Function,
+  onDrop: Function,
+  onDropAccepted: Function,
+  onDropRejected: Function,
+  onDragStart: Function,
+  onDragEnter: Function,
+  onDragOver: Function,
+  onDragLeave: Function,
+
+  children: any, // Contents of the dropzone
+  style: Object, // CSS styles to apply
+  activeStyle: Object, // CSS styles to apply when drop will be accepted
+  rejectStyle: Object, // CSS styles to apply when drop will be rejected
+  className: string, // Optional className
+  activeClassName: string, // className for accepted state
+  rejectClassName: string, // className for rejected state
+
+  disablePreview: boolean, // Enable/disable preview generation
+  disableClick: boolean, // Disallow clicking on the dropzone container to open file dialog
+  onFileDialogCancel: Function, // Provide a callback on clicking the cancel button of the file dialog
+
+  inputProps: Object, // Pass additional attributes to the <input type="file"/> tag
+  multiple: boolean, // Allow dropping multiple files
+  accept: string, // Allow specific types of files. See https://github.com/okonet/attr-accept for more information
+  name: string, // name attribute for the input tag
+  maxSize: number,
+  minSize: number
+};
+
+
 const supportMultiple = (typeof document !== 'undefined' && document && document.createElement) ?
   'multiple' in document.createElement('input') :
   true;
 
 class Dropzone extends React.Component {
-  constructor(props, context) {
+  props: Props;
+
+  static defaultProps: {
+    disablePreview: boolean,
+    disableClick: boolean,
+    multiple: boolean,
+    maxSize: number,
+    minSize: number
+  };
+
+  state: {
+    isDragActive: boolean,
+    isDragReject?: boolean
+  };
+
+  fileInputEl: Object;
+  isFileDialogActive: boolean;
+  enterCounter: number;
+
+  constructor(props: Object, context: Object) {
     super(props, context);
-    this.onClick = this.onClick.bind(this);
-    this.onDragStart = this.onDragStart.bind(this);
-    this.onDragEnter = this.onDragEnter.bind(this);
-    this.onDragLeave = this.onDragLeave.bind(this);
-    this.onDragOver = this.onDragOver.bind(this);
-    this.onDrop = this.onDrop.bind(this);
-    this.onFileDialogCancel = this.onFileDialogCancel.bind(this);
-    this.fileAccepted = this.fileAccepted.bind(this);
+    (this:any).onClick = this.onClick.bind(this);
```

Comment:
you need to define `onClick` as a property type in the class body, to make this flow understand the `this.onClick`...